### PR TITLE
Apply skipping certification verification to authorization as well

### PR DESCRIPTION
**Pull request by ktock**
Fixes: #278

This commit fixes the issue that skipping certificate verification didn't applied to the access to the authorization server.
The behaviour that propagates skip-verification config to the authorizer is compatible to [the current behaviour of `ctr`](https://github.com/containerd/containerd/blob/main/cmd/ctr/commands/resolver.go#L95-L114).

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Can we have a test like this? https://github.com/containerd/nerdctl/blob/f6f48cd29548226c2102f65c26bb7c807f0eca8d/cmd/nerdctl/push_test.go#L121

Can be a separate PR though

File: pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go [hunk 1]
```diff
@@ -77,15 +80,7 @@ func New(refHostname string, optFuncs ...Opt) (remotes.Resolver, error) {
...
-		tr := &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
-		}
-		client := &http.Client{
-			Transport: tr,
-		}
-		regOpts = append(regOpts, docker.WithClient(client))
+		regOpts = append(regOpts, docker.WithClient(newInsecureClient()))
```
Comment (AkihiroSuda) [inline pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go:83]: No need to call `newInsecureClient()` twice

**Comment by ktock** [comment]
> Can we have a test like this?
> https://github.com/containerd/nerdctl/blob/f6f48cd29548226c2102f65c26bb7c807f0eca8d/cmd/nerdctl/push_test.go#L121
> Can be a separate PR though

I'll work on it. We'll need to host an authorization server in CI (maybe we can use https://github.com/cesanta/docker_auth).

**Comment by AkihiroSuda** [comment]
@ktock Do you plan to work on tests before merging this PR or after?

**Comment by ktock** [comment]
Sorry for the late. I add the test in this week but please merge it if this patch is needed urgently.

**Pull request merged**